libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFRootViewManager.m

```diff
@@ -165,9 +179,16 @@ - (void)popViewController:(UIViewController *)viewController
                             [delegate rootViewManager:strongSelf didPopViewControler:viewController];
                         }
                     }];
+                    if(strongSelf->_modalViewController) {
```

Comment:
I would say that this action needs to preclude calls to `[SFRootViewManagerDelegate rootViewManager:didPopViewControler:]`, because that call is really meant to represent the logical conclusion of the root view manager process.  This should be refactored to reflect the intention of the code above it.